nemo/collections/asr/parts/utils/online_clustering.py

```diff
@@ -35,63 +35,63 @@
 
 import numpy as np
 import torch
-from scipy.optimize import linear_sum_assignment
-from sklearn.metrics.pairwise import linear_kernel
-from sklearn.preprocessing import OneHotEncoder
 
 from nemo.collections.asr.parts.utils.offline_clustering import (
     NMESC,
     SpectralClustering,
     getAffinityGraphMat,
     getCosAffinityMatrix,
 )
+from nemo.collections.asr.parts.utils.optimization_utils import linear_sum_assignment
 
 
-def hungarian_algorithm(
-    spk_count: int, U_set: List[int], cmm_P: torch.Tensor, cmm_Q: torch.Tensor, PmQ: List[int], QmP: List[int]
-) -> np.array:
+def get_lsa_speaker_mapping(
+    U_set: torch.Tensor, cmm_P: torch.Tensor, cmm_Q: torch.Tensor, PandQ: torch.Tensor
+) -> torch.Tensor:
     """
     Find a mapping that minimizes the matching cost between the label P and Q.
     One-hot encodding is employed to represent sequence and calculate the cost.
 
     Args:
-        spk_count (int):
-            Estimated speaker count
         U_set (list):
             Whole set of the estimated speakers
         cmm_P (Tensor):
             Length-matched old sequence
         cmm_Q (Tensor):
             Length-matched new sequence
-        PmQ (list):
-            Set P - Q (Difference of sets)
-        QmP (list):
-            Set Q - P (Difference of sets)
+        PandQ (Tensor):
+            Tensor containing the indices of the speakers that are in both old and new sequences
 
     Returns:
         mapping_array (np.array):
             Mapped labels that minimizes the cost
     """
-    enc = OneHotEncoder(handle_unknown='ignore')
     all_spks_labels = [[x] for x in range(len(U_set))]
-    enc.fit(all_spks_labels)
-    enc_P = enc.transform(cmm_P.reshape(-1, 1)).toarray()
-    enc_Q = enc.transform(cmm_Q.reshape(-1, 1)).toarray()
-    stacked = np.hstack((enc_P, enc_Q))
-    cost = -1 * linear_kernel(stacked.T)[spk_count:, :spk_count]
-    row_ind, col_ind = linear_sum_assignment(cost)
+    PandQ_list: List[int] = [int(x.item()) for x in PandQ]
```

Comment:
minor: mentioning dtype in variable name need to be avoided

Reply:
Makes sense since types are strictly annotated for jit script functions.
Fixed.